data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/beacon/GetEth1VotingSummary.java

```diff
@@ -0,0 +1,208 @@
+import static tech.pegasys.teku.infrastructure.http.RestApiConstants.PARAM_STATE_ID;
+import static tech.pegasys.teku.infrastructure.http.RestApiConstants.PARAM_STATE_ID_DESCRIPTION;
+import static tech.pegasys.teku.infrastructure.http.RestApiConstants.RES_INTERNAL_ERROR;
+import static tech.pegasys.teku.infrastructure.http.RestApiConstants.RES_NOT_FOUND;
+import static tech.pegasys.teku.infrastructure.http.RestApiConstants.RES_OK;
+import static tech.pegasys.teku.infrastructure.http.RestApiConstants.TAG_TEKU;
+import static tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition.listOf;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.javalin.http.Context;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiParam;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+import org.apache.commons.lang3.tuple.Pair;
+import org.jetbrains.annotations.NotNull;
+import tech.pegasys.teku.api.ChainDataProvider;
+import tech.pegasys.teku.api.DataProvider;
+import tech.pegasys.teku.api.response.v1.teku.GetEth1VotingSummaryResponse;
+import tech.pegasys.teku.beaconrestapi.MigratingEndpointAdapter;
+import tech.pegasys.teku.infrastructure.json.types.CoreTypes;
+import tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.AsyncApiResponse;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.datastructures.blocks.Eth1Data;
+import tech.pegasys.teku.validator.coordinator.Eth1DataProvider;
+import tech.pegasys.teku.validator.coordinator.Eth1DataProvider.VotingPeriodInfo;
+
+public class GetEth1VotingSummary extends MigratingEndpointAdapter {
+
+  public static final String OAPI_ROUTE = "/teku/v1/beacon/state/:state_id/eth1voting";
+  public static final String ROUTE = routeWithBracedParameters(OAPI_ROUTE);
+
+  private static final SerializableTypeDefinition<Pair<Eth1Data, UInt64>> ETH1DATA_WITH_VOTES_TYPE =
+      SerializableTypeDefinition.<Pair<Eth1Data, UInt64>>object()
+          .name("Eth1DataWithVotes")
+          .withField("eth1_data", Eth1Data.SSZ_SCHEMA.getJsonTypeDefinition(), Pair::getLeft)
+          .withField("votes", CoreTypes.UINT64_TYPE, Pair::getRight)
+          .build();
+
+  private static final SerializableTypeDefinition<Eth1VotingSummary> ETH1VOTING_SUMMARY_TYPE =
+      SerializableTypeDefinition.<Eth1VotingSummary>object()
+          .name("Eth1VotingSummary")
+          .withField(
+              "eth1_data",
```

Comment:
Probably name this `current_eth1_data` to make it clear this is what's currently in the state.

Reply:
Maybe `state_eth1_data` as we could input not the head only?